### ISSUE TITLE
Rule Change: Require `props: true` for `no-param-reassign` options

### ISSUE THREAD

**Issue by xbinaryx** [eslint/eslint#20830]
### What rule do you want to change?

no-param-reassign

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

Other

### Example code

```js
/*eslint no-param-reassign: ["error", { "ignorePropertyModificationsFor": ["foo"] }]*/
```

### What does the rule currently do for this code?

Currently, the rule accepts this configuration without generating any schema validation errors. However, because `props` is not explicitly set to `true`, the `ignorePropertyModificationsFor` list is silently ignored during rule execution.

### What will the rule do after it's changed?

The rule will throw a schema validation error when `ignorePropertyModificationsFor` or `ignorePropertyModificationsForRegex` are provided without explicitly defining `props: true`.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by mdjermanovic** [comment]
Makes sense to me. Since this is breaking change per our [semver policies](https://github.com/eslint/eslint#semantic-versioning-policy) (_Part of the public API is removed or changed in an incompatible way. The public API includes: Rule schemas_ ..."), I'm adding this to the v11.0.0 project for discussion.
